apps/frontend/src/pages/ViewListing.tsx

```diff
@@ -49,21 +58,49 @@ const ViewListing = () => {
     distance: 0,
     images: [{ url: "https://picsum.photos/1200/400" }],
     status: "AVAILABLE",
+    charityId: null,
   });
 
+  const [currentCharity, setCurrentCharity] = useState<CharityObject>();
+
+  const [screenWidth, setScreenSize] = useState<number>(window.innerWidth);
+
   // Load the listing from the api given an ID
   useEffect(() => {
-    _axios_instance
-      .get("/listing/" + id)
-      .then((response) => {
+    const getListing = async () => {
+      try {
+        const response = await _axios_instance.get("/listing/" + id);
         setListingObject(response.data);
         setListingReceived(true);
-      })
-      .catch((error) => {
-        console.log(error);
-      });
+        if (response.data.charityId) {
+          try {
+            const responseCharity =
+              await _axios_instance.get("/charities/current");
+            setCurrentCharity(responseCharity.data);
+            setListingReceived(true);
+          } catch (error) {
+            console.log(error);
+          }
+        }
+      } catch (error) {
+        console.error(error);
+      }
+    };
+
+    //Linter is upset if I don't have this
+    getListing().then();
```

Comment:
You shouldn't need the .then() here. If you really do, then I am upset with the linter

Reply:
It's not an error but a warning about a uncaught promise... 🤷‍♂️